.github/workflows/build.yml

```
@@ -1,5 +1,5 @@
 name: Main
-on: [push]
+on: [push, pull_request]
```

This would trigger this pipeline twice on pull_request push 

How about:

```suggestion
on:
  push:
    branches:
      - master
  pull_request:
```